File: backend/api/models/supply.py
# backend/api/models/supply.py
from django.db import models
from .base import BaseModel

class Supply(BaseModel):
    """Model for supplies/materials management"""
    
    class UnitMeasure(models.TextChoices):
        UNIT = 'UN', 'Unidade'
        KILOGRAM = 'KG', 'Kilograma' 
        MILLILITER = 'ML', 'Mililitro'

    class SupplyType(models.TextChoices):
        VEHICLE = 'VEI', 'Veículo'
        WEAPON = 'ARM', 'Armamento'
        MATERIAL = 'MAT', 'Material'
        UNIFORM = 'UNI', 'Uniforme'

    name = models.CharField('Nome', max_length=200)
    nick_name = models.CharField('Apelido', max_length=100, null=True, blank=True)
    ean_code = models.CharField('Código EAN', max_length=13, null=True, blank=True, unique=True)
    description = models.TextField('Descrição', null=True, blank=True)
    unit_measure = models.CharField(
        'Unidade de Medida',
        max_length=2,
        choices=UnitMeasure.choices,
        default=UnitMeasure.UNIT
    )
    type = models.CharField(
        'Tipo',
        max_length=3,
        choices=SupplyType.choices,
        default=SupplyType.MATERIAL
    )

    class Meta:
        ordering = ['name']
        verbose_name = 'Supply'
        verbose_name_plural = 'Supplies'
        db_table = 'supplies'

    def __str__(self):
        return f"{self.name} ({self.get_type_display()})"

File: frontend/src/types/supply.ts
// src/types/supply.ts
export interface Supply {
    id?: number;
    name: string;
    nick_name?: string;
    ean_code?: string;
    description?: string;
    unit_measure: 'UN' | 'KG' | 'ML';
    unit_measure_display?: string;
    type: 'VEI' | 'ARM' | 'MAT' | 'UNI';
    type_display?: string;
    enabled?: boolean;
    created?: string;
    updated?: string;
}

export const UNIT_MEASURES = [
    { value: 'UN', label: 'Unidade' },
    { value: 'KG', label: 'Kilograma' },
    { value: 'ML', label: 'Mililitro' }
] as const;

export const SUPPLY_TYPES = [
    { value: 'VEI', label: 'Veículo' },
    { value: 'ARM', label: 'Armamento' },
    { value: 'MAT', label: 'Material' },
    { value: 'UNI', label: 'Uniforme' }
] as const;